yaml/com/ibm/VPD/FRUManager.interface.yaml

```diff
@@ -0,0 +1,122 @@
+
+    - name: ReadKeyword
+      description: >
+          Method to read VPD keyword value. Keyword can be read from hardware or
+          from Dbus.
+      parameters:
+          - name: path
+            type: string
+            description: >
+                Dbus/EEPROM Path of the FRU whose keyword needs to be read.
+          - name: data
+            type: variant[struct[string,string],struct[string]]
+            description: >
+                Information required to read value of keyword.
+          - name: target
+            type: boolean
+            description: >
+                Default will be false and will read from hardware in that case,
+                if true will read from cache.
+      returns:
+          - name: value
+            type: array[byte]
+            description: >
+                Value of the given keyword read from the given target.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: GetLocationCode
+      description: >
+          Method to get the expanded location code for the given inventory path.
+          If expanded location code is not available then unexpanded location
+          code will be returned.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU.
+      returns:
+          - name: locationCode
+            type: string
+            description: >
+                Returns the expanded location code if available, else returns
+                the unexpanded location code.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: deleteFRUVPD
+      description: >
+          Method to delete given FRU VPD and set its present property as false.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+    - name: CollectConcurrentlyMaintainableFRUVPD
+      description: >
+          Method to perform VPD collection for the given FRU. This method
+          requires to have configuration JSON. The FRU should have
+          concurrentlyMaintainable flag set to true in JSON to execute this
+          action.
+      parameters:
+          - name: inventoryPath
+            type: object_path
+            description: >
+                Inventory path of the FRU whose VPD needs to be collected.
+      errors:
+          - xyz.openbmc_project.Common.Error.InvalidArgument
+
+properties:
+    - name: VPDCollectionStatus
+      type: enum[self.VPDStatus]
+      default: NotStarted
+      description: >
+          Property which shows the status of the VPD collection for all the FRUs
+          of the system.
+
+enumerations:
+    - name: VPDStatus
+      description: >
+          VPD collection status for all the FRUs of the system.
```

Comment:
Status of VPD collection process for the system?

Reply:
updated